service/controller/v21/resource/scalingstatus/resource.go

```diff
@@ -0,0 +1,169 @@
+		}
+
+		workerASGName, err = controllerCtx.CloudFormation.GetOutputValue(stackOutputs, key.WorkerASGKey)
+		if cloudformationservice.IsOutputNotFound(err) {
+			// Since we are transitioning between versions we will have situations in
+			// which old clusters are updated to new versions and miss the ASG name in
+			// the CF stack outputs. We stop resource reconciliation at this point to
+			// reconcile again at a later point when the stack got upgraded and
+			// contains the ASG name.
+			//
+			// TODO remove this condition as soon as all guest clusters in existence
+			// obtain a ASG name.
+			r.logger.LogCtx(ctx, "level", "debug", "message", "did not find the guest cluster worker ASG name in the cloud formation stack")
+			r.logger.LogCtx(ctx, "level", "debug", "message", "the guest cluster main stack is not upgraded to the newest version yet")
+			r.logger.LogCtx(ctx, "level", "debug", "message", "canceling resource")
+			return nil
+
+		} else if err != nil {
+			return microerror.Mask(err)
+		}
+
+		r.logger.LogCtx(ctx, "level", "debug", "message", "found the guest cluster worker ASG name in the cloud formation stack")
+	}
+
+	var desiredCapacity int
+	{
+		r.logger.LogCtx(ctx, "level", "debug", "message", "finding out desired value in worker asg")
+
+		i := &autoscaling.DescribeAutoScalingGroupsInput{
+			AutoScalingGroupNames: []*string{
+				&workerASGName,
+			},
+		}
+		o, err := controllerCtx.AWSClient.AutoScaling.DescribeAutoScalingGroups(i)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		if len(o.AutoScalingGroups) == 0 {
+			return microerror.Maskf(notFoundError, "asg for name %s", workerASGName)
+		}
+
+		if o.AutoScalingGroups[0].DesiredCapacity == nil {
+			return microerror.Maskf(notFoundError, "desired capacity for asg is nil")
+		}
+
+		desiredCapacity = int(*o.AutoScalingGroups[0].DesiredCapacity)
+
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("found out desired value in worker asg: %d", desiredCapacity))
+	}
+
+	{
+		r.logger.LogCtx(ctx, "level", "debug", "message", "fetching latest version of custom resource")
+
+		newObj, err := r.g8sClient.ProviderV1alpha1().AWSConfigs(customResource.GetNamespace()).Get(customResource.GetName(), metav1.GetOptions{})
+		if err != nil {
+			return microerror.Mask(err)
+		}
+		customResource = *newObj
+
+		r.logger.LogCtx(ctx, "level", "debug", "message", "fetched latest version of custom resource")
+	}
+
+	{
+		r.logger.LogCtx(ctx, "level", "debug", "message", "updating CR status")
+
+		customResource.Status.Cluster.Scaling.DesiredCapacity = desiredCapacity
+
+		_, err = r.g8sClient.ProviderV1alpha1().AWSConfigs(customResource.Namespace).UpdateStatus(&customResource)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		r.logger.LogCtx(ctx, "level", "debug", "message", "updated CR status")
+
+		r.logger.LogCtx(ctx, "level", "debug", "message", "canceling reconciliation")
+		reconciliationcanceledcontext.SetCanceled(ctx)
```

Comment:
I think the problem here might not bee too critical because the capacities do not change all the time. They change sometimes, but stop changing. Once the changing stops the other resources are free to go ahead again. 

I noticed that problem a while ago as well, but in another context. Pretty interesting. I came up with a not so perfect solution in the `statusresource`. There I tried to update the heartbeats after some fixed period of time. See [`isHeartBeatOverdue`](https://github.com/giantswarm/statusresource/pull/22/files#diff-f7859333652d5fe563fd2a97e6e0704fR189) in my PR. That is definitely something we should think about more. I am not really sure how this should be in a perfect world.